siad/gateway_test.go

```diff
@@ -28,8 +28,11 @@ func (dt *daemonTester) addPeer() *daemonTester {
 	for len(dt.gateway.Info().Peers) != len(newPeer.gateway.Info().Peers) {
 		time.Sleep(time.Millisecond)
 	}
+	// force synchronization to dt, in case newPeer tried to synchronize to
+	// one of dt's peers.
 	for dt.state.Height() != newPeer.state.Height() {
-		time.Sleep(time.Millisecond)
+		newPeer.gateway.Synchronize()
+		time.Sleep(10 * time.Millisecond)
```

Comment:
that sounds pretty reasonable. Selecting a random peer I don't think is the best strategy anyway, you probably want some priority setup or something.